### PR TITLE
Added return values to rascal commands executed from vscode.

### PR DESCRIPTION
**Pull request by PieterOlivier**
Rascal commands executed from typescript over the lsp bridge could can now return values to the typescript side.
Return values are serialized to json so they are easy to work with in typescript.

### CODE REVIEW

File: rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParametricTextDocumentService.java [hunk 1]
```diff
@@ -357,10 +359,6 @@ private ParametricFileFacts facts(TextDocumentIdentifier doc) {
         return facts(doc.getUri());
     }
 
-    private ParametricFileFacts facts(TextDocumentItem doc) {
```
Comment (DavyLandman) [inline rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParametricTextDocumentService.java:360]: while I get that this one is removed (most likely due to nobody using it). For the next time I have a preference to keep cleanup in separate PRs.

**Review by DavyLandman — COMMENTED** [review]
I like it, I think @jurgenvinju should review it. Since @PieterOlivier and I worked together a bit on this.

But I like this solution for #168 since this was just missing about the command pattern.

Also: it's not truly optional, as there is no "null" for `value` in rascal.

File: rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParametricTextDocumentService.java (hunk 1 above)
Comment (PieterOlivier) [reply rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParametricTextDocumentService.java]: I have reverted all changes not related to core purpose of this PR.

**Review by jurgenvinju — APPROVED** [review]
Perfect extension :-) Thanks! We have to document somewhere (let's wait until rascal2doc is released with the rascal-maven-plugin, that normally _nothing_ is done with the return value.
